scripts/compare-statements.js

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+'use strict';
+
+/**
+ *
+ * Sort a list of compatibility statements based upon reverse-chronological order in the following order:
+ 1. Statements with only version added and/or removed (and potentially notes)
+ 2. Statements with alternative names or prefixes
+ 3. Statements with partial support
+ 4. Statements with flags
+ *
+ */
+
+const compareStatements = (a, b) => {
+  let has = {
+    a: {
+      flags: a.flags != undefined,
+      altname: a.prefix != undefined || a.alternative_name != undefined,
+      partial: a.partial_implementation != undefined,
+    },
+    b: {
+      flags: b.flags != undefined,
+      altname: b.prefix != undefined || b.alternative_name != undefined,
+      partial: b.partial_implementation != undefined,
+    },
+  };
+
+  if (has.a.flags && !has.b.flags) return 1;
+  if (!has.a.flags && has.b.flags) return -1;
+
+  if (has.a.altname && !has.b.altname) return 1;
+  if (!has.a.altname && has.b.altname) return -1;
+
+  if (has.a.partial && !has.b.partial) return 1;
+  if (!has.a.partial && has.b.partial) return -1;
+
+  if (typeof a.version_added == 'string' && typeof b.version_added == 'string')
+    return Number(a.version_added) < Number(b.version_added) ? 1 : -1;
```

Comment:
Can we use compare-versions here instead? This doesn't work with Node.js versions, as `Number('17.0.0')` is NaN.